File: solutions/cpp/etl/1/etl.cpp
#include "etl.h"

using namespace std;

#include <string>

namespace etl {

map<char, int> transform(map<int, vector<char>> old){
    map<char, int> res;
    for( map<int,vector<char>>::iterator iter=old.begin(); iter!=old.end(); ++iter){
        for(char ch : iter->second){
            res.insert({tolower(ch), iter->first}); 
        }
    }
    return res;
}

}  // namespace etl

File: solutions/javascript/etl/1/etl.js
// ETL exercise first version: loops

export const transform = (old) => {
  let new_format = {};
  for(let value in old) {
    for(let letter of old[value]){
      new_format[letter.toLowerCase()] =  parseInt(value);
    }
  }
  return new_format;
};
